lib/mix/lib/mix/tasks/deps.compile.ex

```diff
@@ -33,14 +33,18 @@ defmodule Mix.Tasks.Deps.Compile do
   import Mix.Dep, only: [loaded: 1, available?: 1, loaded_by_name: 2,
                          make?: 1, mix?: 1]
 
-  @switches [include_children: :boolean]
+  @switches [include_children: :boolean, force: :boolean]
 
   @spec run(OptionParser.argv) :: :ok
   def run(args) do
     unless "--no-archives-check" in args do
       Mix.Task.run "archive.check", args
     end
 
+    if "--force" in args do
```

Comment:
Yes, it is better to parse force in OptionParser and pass it as an option to `compile`.